### PR TITLE
[restc-cpp] make using boost-log an optional feature

### PR DESCRIPTION
**Pull request by sandro97git**
This disables logging with boost-log by default and makes it an optional feature. This also solves the problem that this library configures boost-log to link dynamically which leads to problems on linux where vcpkg links statically by default.

### CODE REVIEW

File: ports/restc-cpp/portfile.cmake [hunk 1]
```diff
@@ -14,6 +14,7 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     openssl       RESTC_CPP_WITH_TLS
```
Comment (NancyLi1013) [inline ports/restc-cpp/portfile.cmake:14]:
```suggestion
    FEATURES 
        openssl       RESTC_CPP_WITH_TLS
```

File: ports/restc-cpp/vcpkg.json [hunk 2]
```diff
@@ -21,6 +21,12 @@
     "zlib"
   ],
   "features": {
```
Comment (NancyLi1013) [inline ports/restc-cpp/vcpkg.json:23]: Could you please also update version in this files?

Add `"port-version": 1,` below  `"version-semver": "0.10.0",`. Namely like this:

```
"version-semver": "0.10.0",
"port-version": 1,
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Have you tested this feature `boost-log`? Does it work fine?

**Comment by sandro97git** [comment]
> Have you tested this feature `boost-log`? Does it work fine?
yes it works as intended

**Comment by sandro97git** [comment]
Any hints on how I can fix the checks?

**Comment by NancyLi1013** [comment]
Please run `vcpkg x-add-version --overwrite-version restc-cpp` to solve the failures on x86-windows.

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your PR @sandro97git.

**Review by vicroms — CHANGES_REQUESTED** [review]
Remove `boost-log`  from the main list of dependencies

```
  "dependencies": [
    "boost-chrono",
    "boost-context",
    "boost-coroutine",
    "boost-date-time",
    "boost-filesystem",
    "boost-log",
    "boost-program-options",
    "boost-system",
    "boost-uuid",
    "rapidjson"
  ],
  ```

**Comment by NancyLi1013** [comment]
It should be fine now, thanks for your review @vicroms.

**Comment by vicroms** [comment]
missing a `boost-asio` dependency that was previously transitively supplied by `boost-log`

**Pull request merged**